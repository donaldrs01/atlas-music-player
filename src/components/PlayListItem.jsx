import React from 'react';

const PlayListItem = ({ title, artist, length }) => {
    return (
        <div className={`flex justify-between items-center w-full py-2 cursor-pointer hover:bg-gray-100 hover:text-gray-700`}
        >
            <div>
                <p className="text-xl font-bold text-gray-700">{title}</p>
                <p className="text-l text-gray-500">{artist}</p>
            </div>
            <div className="pr-6">
                <p className="text-l text-gray-400">{length}</p>
            </div>
        </div>
    );
};

export default PlayListItem;