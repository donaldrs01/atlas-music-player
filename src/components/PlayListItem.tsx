import React from "react";

type PlayListItemProps = {
  title: string;
  artist: string;
  length: number;
  isPlaying: boolean;
  onClick: () => void;
};

const formatTime = (seconds: number) => {
  const minutes = Math.floor(seconds / 60); // Retrieve whole number of minutes
  const secs = seconds % 60; // Calculates remaining seconds
  return `${minutes}:${secs < 10 ? "0" : ""}${secs}`; // Seconds always has two digits (leading 0 if < 10)
};

const PlayListItem: React.FC<PlayListItemProps> = ({
  title,
  artist,
  length,
  isPlaying,
  onClick,
}) => {
  return (
    <div
      onClick={onClick}
      className={`group flex w-full cursor-pointer items-center justify-between p-2 ${isPlaying ? "bg-rose-500 font-bold text-white" : "bg-chartreuse-50 hover:bg-rose-200"} `}
    >
      <div className="flex flex-col">
        <p
          className={`text-lg ${isPlaying ? "text-white" : "text-rose-300"} group-hover:text-rose-50`}
        >
          {title}
        </p>
        <p
          className={`text-sm ${isPlaying ? "text-white" : "text-rose-200"} group-hover:text-rose-50`}
        >
          {artist}
        </p>
      </div>
      <div className="pr-6">
        <p
          className={`text-sm ${isPlaying ? "text-white" : "text-rose-200"} group-hover:text-rose-50`}
        >
          {formatTime(length)}
        </p>
      </div>
    </div>
  );
};

export default PlayListItem;
