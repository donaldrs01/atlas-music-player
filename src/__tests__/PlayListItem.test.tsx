import { render } from "@testing-library/react";
import PlayListItem from "../components/PlayListItem";
import { describe, test, expect, vi } from "vitest";

describe("PlayListItem Snapshot Tests", () => {
  const mockOnClick = vi.fn();

  test("correctly displays title, artist and length of current track", () => {
    const { container } = render(
      <PlayListItem
        title="Trains"
        artist="Porcupine Tree"
        length={257}
        isPlaying={false}
        onClick={mockOnClick}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("displays correct song information for another song", () => {
    const { container } = render(
      <PlayListItem
        title="Goliath"
        artist="Karnivool"
        length={331}
        isPlaying={false}
        onClick={mockOnClick}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("displays a song with no artist provided", () => {
    const { container } = render(
      <PlayListItem
        title="Blue Sky"
        length={200}
        isPlaying={false}
        onClick={mockOnClick}
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test("displays a song with no length provided", () => {
    const { container } = render(
      <PlayListItem
        title="Comfortably Numb"
        artist="Pink Floyd"
        isPlaying={true}
        onClick={mockOnClick}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
