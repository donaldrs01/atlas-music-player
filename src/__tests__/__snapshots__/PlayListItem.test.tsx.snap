// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayListItem Snapshot Tests > correctly displays title, artist and length of current track 1`] = `
<div>
  <div
    class="group flex w-full cursor-pointer items-center justify-between p-2 bg-light-listItem hover:bg-light-hover dark:bg-dark-bg dark:hover:bg-dark-hover"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-lg text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Trains
         
      </p>
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Porcupine Tree
         
      </p>
    </div>
    <div
      class="text-light-accent dark:text-dark-accent pr-6 group-hover:text-light-primary dark:group-hover:text-dark-primary"
    >
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        4:17
         
      </p>
    </div>
  </div>
</div>
`;

exports[`PlayListItem Snapshot Tests > displays a song with no artist provided 1`] = `
<div>
  <div
    class="group flex w-full cursor-pointer items-center justify-between p-2 bg-light-listItem hover:bg-light-hover dark:bg-dark-bg dark:hover:bg-dark-hover"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-lg text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Blue Sky
         
      </p>
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
         
      </p>
    </div>
    <div
      class="text-light-accent dark:text-dark-accent pr-6 group-hover:text-light-primary dark:group-hover:text-dark-primary"
    >
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        3:20
         
      </p>
    </div>
  </div>
</div>
`;

exports[`PlayListItem Snapshot Tests > displays a song with no length provided 1`] = `
<div>
  <div
    class="group flex w-full cursor-pointer items-center justify-between p-2 bg-light-now-playing dark:bg-dark-now-playing font-bold text-white"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-lg text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Comfortably Numb
         
      </p>
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Pink Floyd
         
      </p>
    </div>
    <div
      class="text-light-accent dark:text-dark-accent pr-6 group-hover:text-light-primary dark:group-hover:text-dark-primary"
    >
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        NaN:NaN
         
      </p>
    </div>
  </div>
</div>
`;

exports[`PlayListItem Snapshot Tests > displays correct song information for another song 1`] = `
<div>
  <div
    class="group flex w-full cursor-pointer items-center justify-between p-2 bg-light-listItem hover:bg-light-hover dark:bg-dark-bg dark:hover:bg-dark-hover"
  >
    <div
      class="flex flex-col"
    >
      <p
        class="text-lg text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Goliath
         
      </p>
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        Karnivool
         
      </p>
    </div>
    <div
      class="text-light-accent dark:text-dark-accent pr-6 group-hover:text-light-primary dark:group-hover:text-dark-primary"
    >
      <p
        class="text-sm text-light-primary dark:text-dark-primary group-hover:text-light-primary dark:group-hover:text-dark-primary"
      >
         
        5:31
         
      </p>
    </div>
  </div>
</div>
`;
